src/clj_kondo/impl/analyzer.clj

```diff
@@ -886,6 +911,8 @@
         varargs-min-arity (when arities (get-in arities [:varargs :min-arity]))
         arglist-strs (when arities (into [] (keep :arglist-str) (vals arities)))
         parsed-bodies (mapcat :parsed parsed-bodies)]
+    (when-let [kind (def-fn? ctx)]
```

Comment:
I would put the linter-disabled? call here so you can avoid the call to `def-fn?` when it's not necessary